### DISCUSSION TITLE
Hot wo get aligned data frame output table for Julia DataFrames

### DISCUSSION THREAD

**Discussion by mucahitzor** [quarto-dev/quarto-cli#6097] [Q&A] [answered]
### Description

Hello everyone! I have been using Julia for some time and have tried different ways to code. So far I liked Pluto, but wanted to try something else as well. Since I am coming from R, I wanted to try Quarto with Julia in VSCode. One problem I am having is that when I render for HTML, data frame outputs look not as I expected:
[default output](https://global.discourse-cdn.com/business5/uploads/julialang/original/3X/4/4/442c6b3e18b042256105049363861ad441b4ca4f.png)

Quarto has some options for tables such as tbl-colwidths, and df-print which in my case didn’t work. I guess this is probably because they are not defined for Julia? I don’t know but I was able to overcome this using markdown_table function, but this output lacks information about the types of columns.

[markdown output](https://global.discourse-cdn.com/business5/uploads/julialang/original/3X/4/c/4c9e02898dd4f1dd7ca92e2cd13c3355a943277e.png)

Is there any way that I can get the first table fixed? Thanks!

Here is how my .qmd looks like:

````markdown
---
title: "Data Frames" 
execute:
    cache: true
jupyter: julia-1.9
---

```{julia}
using DataFramesMeta, MarkdownTables, Random, Distributions
```

```{julia}
y = DataFrame(
    x = round.(rand(10),digits=2),
    y = rand(["A","B"],10),
    z = sample(1:100,10),
    w = 1
)
```

```{julia}
y = DataFrame(
    x = round.(rand(10),digits=2),
    y = rand(["A","B"],10),
    z = sample(1:100,10),
    w = 1
)

markdown_table(y)
```
````

**Comment by cscheid** [comment]
> Quarto has some options for tables such as tbl-colwidths, and df-print which in my case didn’t work.

These are `knitr` options rather than `quarto` options, and so they will generally only work in the `knitr` engine.

(We'd like to get better table printing in non-knitr formats, and we agree that the options aren't great now.)

**Reply by mcanouil** [reply, marked as answer]
A workaround would be to use `knitr` as the engine and use `JuliaCall` (<https://non-contradiction.github.io/JuliaCall/index.html>) to call Julia.

**Reply by mucahitzor** [reply]
Thank you very much to both of you.
